tests/ui_tools/test_boxes.py

```diff
@@ -60,3 +60,27 @@ def test_autocomplete_mentions(self, write_box, users_fixture,
         write_box.view.users = users_fixture
         typeahead_string = write_box.autocomplete_mentions(text, state, index)
         assert typeahead_string == required_typeahead
+
+    @pytest.mark.parametrize('text, state, required_typeahead', [
+        ('#Stream', 0, '#**Stream 1**'),
+        ('#Stream', 1, '#**Stream 2**'),
+        ('#S', 0, '#**Some general stream**'),
+        ('#S', 1, '#**Secret stream**'),
+        ('#S', 2, '#**Stream 1**'),
+        ('#S', 3, '#**Stream 2**'),
+        ('#So', 0, '#**Some general stream**'),
+        ('#So', 1, None),
+        ('#Se', 0, '#**Secret stream**'),
+        ('#Se', 1, None),
+        ('#St', 0, '#**Stream 1**'),
+        ('#St', 1, '#**Stream 2**'),
+        ('#Stream 1', 0, '#**Stream 1**'),
+        ('No match', 0, None)
+    ])
+    def test_autocomplete_streams(self, write_box, streams_fixture,
+                                  text, state, required_typeahead):
+        write_box.view.pinned_streams = [[stream['name']]
+                                         for stream in streams_fixture]
+        write_box.view.unpinned_streams = []
```

Comment:
The unpinned streams being empty feels a little suspicious. I know it would affect the search order, but perhaps just split them up into the two types for now, eg. Stream 1/2 into unpinned, and that would mean the order would be the same? Or even put Stream 1/2 into pinned, to show how they come first?

@amanagr This makes me wonder if we should encode some streams as pinned by default in the streams fixture(s)?

Reply:
Done. :+1: